### PR TITLE
[14.0][IMP] sale_order_type: Commits from v13

### PR DESCRIPTION
**Pull request by sergio-teruel**
This PR adds two commits from v13.0

cc @Tecnativa

https://github.com/OCA/sale-workflow/commit/729afc9b3165723aea0f198e70dc9cb3f9af95be
4d1d77f860a9bc33542369138dc2d49ced84cf75

### CODE REVIEW

File: sale_order_type/models/sale.py [hunk 1]
```diff
@@ -38,9 +41,11 @@ def _compute_sale_type_id(self):
                 )
             else:
                 sale_type = (
-                    record.partner_id.with_company(record.company_id).sale_type
-                    or record.partner_id.commercial_partner_id.with_company(
-                        record.company_id
+                    record.partner_id.with_context(
+                        force_company=record.company_id.id
```
Comment (rousseldenis) [inline sale_order_type/models/sale.py]: Could you change this to with_company and check runbot warnings ?

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Comment by sergio-teruel** [comment]
@rousseldenis changes done!!

The runbot warning messages was due to sale_order_line_discount_validation module.
I have done a commit fixing it.

**Comment by rousseldenis** [comment]
> @rousseldenis changes done!!
> 
> The runbot warning messages was due to sale_order_line_discount_validation module.
> I have done a commit fixing it.

Yes, but also for the force_company. Thanks

**Comment by sergio-teruel** [comment]
> > @rousseldenis changes done!!
> > The runbot warning messages was due to sale_order_line_discount_validation module.
> > I have done a commit fixing it.
> 
> Yes, but also for the force_company. Thanks

I already done !

**Review by rousseldenis — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Cherry-picked for v15 at 89b310c03322673ad91a70a67b9a64f55a297848 and bca91d3f62556a9fa7056e426d30d40dcc9b9eb0

**Pull request merged**